services/builder/configProcessor.go

```diff
@@ -110,10 +110,11 @@ func configProcessor(c pb.ContextBoxServiceClient, wg *sync.WaitGroup) error {
 			config.CurrentState = tmpConfig.DesiredState
 		}
 		if toDelete != nil {
-			log.Info().Msgf("Deleting nodes for config %s", config.Name)
+			name := config.Name
```

Comment:
Just curious, how is this gonna work? If `config.Name` is `nil`, then `name` will become `nil` as well, won't it?

Reply:
```go
config, err = deleteNodes(config, toDelete)
```
we assign to config here (notice there is not := operator) thus in the `if err != nil` block the config will be nil and panic.

Reply:
the panic happens in the `if err != nil` block when accessing `config.Name` as the function `deleteNodes` returned an error and config will be set to nil.

it only is being changed in this block because there is a return in the `if err != nil` block.